src/test/fuzz/txrequest.cpp

```diff
@@ -0,0 +1,363 @@
+        for (int peer = 0; peer < MAX_PEERS; ++peer) {
+            m_announcements[txhash][peer].m_state = State::NOTHING;
+        }
+        Cleanup(txhash);
+
+        // Call TxRequestTracker's implementation.
+        m_tracker.ForgetTxHash(TXHASHES[txhash]);
+    }
+
+    void ReceivedInv(int peer, int txhash, bool is_wtxid, bool preferred, std::chrono::microseconds reqtime)
+    {
+        // Apply to naive structure: if no announcement for txidnum/peer combination
+        // already, create a new CANDIDATE; otherwise do nothing.
+        Announcement& ann = m_announcements[txhash][peer];
+        if (ann.m_state == State::NOTHING) {
+            ann.m_preferred = preferred;
+            ann.m_state = State::CANDIDATE;
+            ann.m_time = reqtime;
+            ann.m_is_wtxid = is_wtxid;
+            ann.m_sequence = m_current_sequence++;
+            ann.m_priority = m_tracker.ComputePriority(TXHASHES[txhash], peer, ann.m_preferred);
+
+            // Add event so that AdvanceToEvent can quickly jump to the point where its reqtime passes.
+            if (reqtime > m_now) m_events.push(reqtime);
+        }
+
+        // Call TxRequestTracker's implementation.
+        m_tracker.ReceivedInv(peer, GenTxid{is_wtxid, TXHASHES[txhash]}, preferred, reqtime);
+    }
+
+    void RequestedTx(int peer, int txhash, std::chrono::microseconds exptime)
+    {
+        // Apply to naive structure: if a CANDIDATE announcement exists for peer/txhash,
+        // convert it to REQUESTED, and change any existing REQUESTED announcement for the same txhash to COMPLETED.
+        if (m_announcements[txhash][peer].m_state == State::CANDIDATE) {
+            for (int peer2 = 0; peer2 < MAX_PEERS; ++peer2) {
+                if (m_announcements[txhash][peer2].m_state == State::REQUESTED) {
+                    m_announcements[txhash][peer2].m_state = State::COMPLETED;
+                }
+            }
+            m_announcements[txhash][peer].m_state = State::REQUESTED;
+            m_announcements[txhash][peer].m_time = exptime;
+        }
+
+        // Add event so that AdvanceToEvent can quickly jump to the point where its exptime passes.
+        if (exptime > m_now) m_events.push(exptime);
+
+        // Call TxRequestTracker's implementation.
+        m_tracker.RequestedTx(peer, TXHASHES[txhash], exptime);
+    }
+
+    void ReceivedResponse(int peer, int txhash)
+    {
+        // Apply to naive structure: convert anything to COMPLETED.
+        if (m_announcements[txhash][peer].m_state != State::NOTHING) {
+            m_announcements[txhash][peer].m_state = State::COMPLETED;
+            Cleanup(txhash);
+        }
+
+        // Call TxRequestTracker's implementation.
+        m_tracker.ReceivedResponse(peer, TXHASHES[txhash]);
+    }
+
+    void GetRequestable(int peer)
+    {
+        // Implement using naive structure:
+        std::vector<std::tuple<uint64_t, int, bool>> result; //!< list of (sequence number, txhash, is_wtxid) pairs.
```

Comment:
8d0dd46b4fcdf4c664747e638eda046f8d51079c
```suggestion
        std::vector<std::tuple<uint64_t, int, bool>> result; //!< list of (sequence number, txhash, is_wtxid) tuples.
```

Reply:
Done.